AudiusProject/audius-protocol
# [ASI-945] Remove more IPFS functionality

### Description
Creator Node:
- Remove IPFS health check (`/health_check/ipfs`) and its config
  - <strike>Question: do we have any infra that relies on this that needs to be updated before this merges?</strike>
  - Opened [separate PR](https://github.com/AudiusProject/audius-protocol/pull/2795) to remove external consumer of the IPFS health check
- Remove IPFS monitors (read-write status, pin status)
- Remove creator-node’s ipfsClient and its many consumers
- Remove config and functionality for falling back to checking IPFS for content when not found on disk
- Remove IPFS sync
- Remove functionality to pin CIDs
- Fix proxyquire usage leaking stubbed values outside of its intended tests and causing other tests to fail
- Rename:
  - “multihash” -> “CID” (in some but not all functions, variables, and comments)
  - “ipfsAdd” -> “fileHasher”

Mad Dog:
- Remove mad-dog’s ipfsClient, its consumers, and its ipfs-http-client dependency


### Tests
Follow the test plan from the previous remove-ipfs PR -- pasted here for convenience:
#### Testing Plan
1. Checkout the branch and start the services (`cd audius-protocol && git checkout theo-remove-ipfs && A up`)
2. Start the web client (`cd audius-client/packages/web && scp -r remote-dev-instance:.audius ~/ && npm run start:dev:cloud`)
3. Sign up and verify changing your profile photo -- upload a new photo and look for the profile photo hash inside the payload of `/metadata` POST call in the Network tab. Then refresh the page and filter the Network tab for that hash. You should see valid requests for `/ipfs/<your hash>/<dimensions>` and 404s (for legacy purposes) for `/ipfs/<your hash>`
5. Verify uploading and playing track. Find the hash of the segments in the final `/async_processing_status` request and the hash of the cover artwork (the `/tracks/metadata` metadataMultihash and the`/image_upload` dirCID). Click "View track page" and do a hard refresh to make sure you see the right cover image hash loaded. Click Play to make sure you see the segment hashes loaded and hear the right audio

**Before releasing to prod:** push to staging and repeat steps 3-5 there

### How will this change be monitored? Are there sufficient logs?
- Shouldn't see any more logs containing "IPFS Nodes connected + data export received"
- Perform a sanity check on the read and write flows mentioned above in staging and investigate this change if any issues come up
